players/base.py

```diff
@@ -0,0 +1,256 @@
+import logging
+from enum import Enum
+from typing import List
+
+import memory
+import xbox
+
+logger = logging.getLogger(__name__)
+
+
+class PlayerMagicNumbers(Enum):
+    CHAR_STRUCT_SIZE = 0x94
+    CHAR_STAT_POINTER = 0x003AB9B0
+    BATTLE_STRUCT_SIZE = 0x90
+    BATTLE_STATE_STRUCT_SIZE = 0xF90
+    BATTLE_STATE_STRUCT_POINTER = 0x00D334CC
+    CUR_HP = 0x00D32078
+    MAX_HP = 0x00D32080
+    BATTLE_CUR_HP = 0x00F3F7A4
+    BATTLE_MAX_HP = 0x00F3F7A8
+    LUCK = 0x34
+    ACCURACY = 0x36
+    BATTLE_OVERDRIVE = 0x5BC
+    OVERDRIVE = 0x39
+    AFFECTION_POINTER = 0x00D2CABC
+    SLVL = 0x00D32097
+    RNG_COMP = 0x7FFFFFFF
+    ESCAPED = 0xDC8
+    ACTIVE_BATTLE_SLOTS = 0x00F3F76C
+    BACKLINE_BATTLE_SLOTS = 0x00D2C8A3
+    ARMOR_ID = 1
+    WEAPON_ID = 0
+
+
+class Player:
+    def __init__(self, name: str, id: int, battle_menu: List[int]):
+        self.name = name
+        self.id = id
+        self.struct_offset = id * PlayerMagicNumbers.CHAR_STRUCT_SIZE
+        self.char_rng = 20 + id
+        self.battle_menu = battle_menu
+
+    def __eq__(self, other):
+        if isinstance(other, int):
+            return self.id == other
+        else:
```

Comment:
Small Nitpick: In all these places you are doing:

```python
if ...
    return ...
else:
    return ...
```

We ought to be doing:

```python
if ...
    return ...
return ...
```

No need for the `else` since we are returning in the `if` block.

Reply:
While that is true, I think that I like the explicitness of it here - especially since it's a branching path, it makes it clearer that it's one or the other. I'd prefer to not change it, but if you still want me to I will.